### ISSUE TITLE
CA2000 Erroneous warnings for IAsyncDisposable

### ISSUE THREAD

**Issue by symbiogenesis** [DotNetAnalyzers/StyleCopAnalyzers#3889]
Consider the following examples:

```csharp
await using var asyncDisposable = new AsyncDisposable();

Foo();
```

This should **never** throw CA2000, but I see it.

Consider, also:

```csharp
var asyncDisposable = new AsyncDisposable();

await using (asyncDisposable.ConfigureAwait(false))
{
    Foo();
}
```

I would not expect this to result in a warning as long as `asyncDisposable` is not accessed again outside of the `await using` block.

**Note:** In the event of `ConfigureAwait()` being applied, the disposable becomes type `ConfiguredAsyncDisposable`

**Comment by bjornhellander** [comment]
CAxxxx diagnostics are triggered by a package built from this repo:
https://github.com/dotnet/roslyn-analyzers

**Comment by symbiogenesis** [comment]
Ah, thanks!
